server/src/main/java/org/elasticsearch/transport/TcpChannel.java

```diff
@@ -66,6 +66,13 @@ public interface TcpChannel extends CloseableChannel {
      */
     void addConnectListener(ActionListener<Void> listener);
 
+    /**
+     * Report an exception on this channel
+     *
+     * @param e the exception
+     */
+    void onException(Exception e);
```

Comment:
Naming nit, maybe `setCloseException` or something like that? To me `onException` sounds like something that should react to the exception by e.g. closing the channel and that's not what is happening.